### ISSUE TITLE
Vim: improved, more granular styling for common languages

### ISSUE THREAD

**Issue by jan-warchol** [jan-warchol/selenized#68]
As of 76b0a1529d0518c8d9a922f30ee656cc29c53f9f vim theme only specifies styling for basic token groups. More detailed styling rules would be a great improvement. Note: it may be a good idea to use [vim-polyglot](https://github.com/sheerun/vim-polyglot).

Requires #67 to be done first.

Languages that I'm likely to work on first:
* python
* shell
* vimscript

Languages that I'd like to work on but need help from people who know them better:
* Javascript
* ruby
* Java

**Comment by jan-warchol** [comment]
One thing I'm considering is differentiating numbers and booleans from strings in some way.

**Comment by jan-warchol** [comment]
Python exceptions definitely need to be more visible. Currently they are linked to `Type` group, which is green and that doesn't have enough contrast from surrounding cyan and blue (check on [this canonical example](https://github.com/jan-warchol/highlighting-code-samples/blob/master/jan-warchol/limestone_palette_builder.py)).

**Comment by fladd** [comment]
At first, it appeared that Selenized strives to become a simple and better implemented alternative to Solarized (which [has been criticized](https://www.vi-improved.org/loathing/) for its implementation). That impression apparently has also led to a suggestion for shipping Selenized with Vim itself (https://github.com/vim/vim/issues/1665#issuecomment-522303384). Making Selenized reliant on third-party syntax scripts (vim-polyglot) means taking an entirely different route (and probably also means not being considered for being shipped with Vim). This is of course fine. Just something to be aware of.

**Comment by jan-warchol** [comment]
Good point. Let me say that I'm definitely not going to make selenized _dependent_ on vim-polyglot - what I'd like to see is selenized being _able_ to take advantage of such plugin if the user has it installed.

**Comment by memco** [comment]
I support deviating to enhance the experience. Personally, I'm always evaluating themes based on how they highlight punctuation like separators, string identifiers, etc. Obviously there should be a sane default baseline, but as the theme progresses adding more specialization is welcome IMO.

**Comment by jsit** [comment]
I would refrain from using polyglot, as not everybody has it installed. Instead, I'd suggest loading up a bunch of filetypes in a `vim --clean` session, then do `:highlight` to see what highlight groups are defined by vanilla Vim out of the box, and target those.

**Comment by dngray** [comment]
I somewhat agree with @fladd, personally I don't like to see vim depend on a whole lot of third party plugins. I try to keep what I use to a minimum, which I think is prudent for security.
